packages/store-indexer/bin/postgres-indexer.ts

```diff
@@ -0,0 +1,137 @@
+
+const publicClient = createPublicClient({
+  chain,
+  transport: fallback(
+    // If one or more RPC URLs are provided, we'll configure the transport with only those RPC URLs
+    transports.length > 0
+      ? transports
+      : // Otherwise use the chain defaults
+        [webSocket(), http()]
+  ),
+  pollingInterval: 1000,
+});
+
+// Fetch the chain ID from the RPC if no chain object was found for the provided chain ID.
+// We do this to match the downstream logic, which also attempts to find the chain ID.
+const chainId = chain?.id ?? (await publicClient.getChainId());
+
+const database = drizzle(postgres(env.DATABASE_URL), {
+  logger: new DefaultLogger(),
+});
+
+let startBlock = env.START_BLOCK;
+
+const storageAdapter = await postgresStorage({ database, publicClient });
+
+// Resume from latest block stored in DB. This will throw if the DB doesn't exist yet, so we wrap in a try/catch and ignore the error.
+try {
+  const currentChainStates = await database
+    .select()
+    .from(storageAdapter.internalTables.chain)
+    .where(eq(storageAdapter.internalTables.chain.chainId, chainId))
+    .execute();
+  // TODO: replace this type workaround with `noUncheckedIndexedAccess: true` when we can fix all the issues related (https://github.com/latticexyz/mud/issues/1212)
+  const currentChainState: (typeof currentChainStates)[number] | undefined = currentChainStates[0];
+
+  if (currentChainState != null) {
+    if (currentChainState.schemaVersion != schemaVersion) {
+      console.log(
+        "schema version changed from",
+        currentChainState.schemaVersion,
+        "to",
+        schemaVersion,
+        "cleaning database"
+      );
+      await cleanDatabase(database);
+    } else if (currentChainState.lastUpdatedBlockNumber != null) {
+      console.log("resuming from block number", currentChainState.lastUpdatedBlockNumber + 1n);
+      startBlock = currentChainState.lastUpdatedBlockNumber + 1n;
+    }
+  }
+} catch (error) {
+  // ignore errors, this is optional
+}
+
+const { latestBlockNumber$, blockStorageOperations$ } = await createStoreSync({
+  storageAdapter,
+  publicClient,
+  startBlock,
+  maxBlockRange: env.MAX_BLOCK_RANGE,
+});
+
+combineLatest([latestBlockNumber$, blockStorageOperations$])
+  .pipe(
+    filter(
+      ([latestBlockNumber, { blockNumber: lastBlockNumberProcessed }]) => latestBlockNumber === lastBlockNumberProcessed
+    ),
+    first()
+  )
+  .subscribe(() => {
+    console.log("all caught up");
+  });
+
+// @see https://fastify.dev/docs/latest/
+const server = fastify({
+  maxParamLength: 5000,
+});
+
+await server.register(import("@fastify/cors"));
+
+// @see https://trpc.io/docs/server/adapters/fastify
+server.register(fastifyTRPCPlugin<AppRouter>, {
+  prefix: "/trpc",
+  trpcOptions: {
+    router: createAppRouter(),
+    createContext: async () => ({
+      queryAdapter: await createQueryAdapter(database),
+    }),
+  },
+});
+
+await server.listen({ port: env.PORT });
```

Comment:
should we make the host configurable here or do you want to update that for both sqlite and postgres indexer in a followup PR? (like https://github.com/latticexyz/mud/pull/1219/commits/e70bd42836557c36f0b97b589a187fe3f55e1e5b)

Reply:
oh yep, can add that here